Code/EnginePlugins/OpenXRPlugin/OpenXRSingleton.cpp

```diff
@@ -0,0 +1,1122 @@
+
+  return m_hColorRT;
+}
+
+void ezOpenXR::GALDeviceEventHandler(const ezGALDeviceEvent& e)
+{
+  if (e.m_Type == ezGALDeviceEvent::Type::BeforeBeginFrame)
+  {
+    BeforeBeginFrame();
+  }
+}
+
+void ezOpenXR::GameApplicationEventHandler(const ezGameApplicationExecutionEvent& e)
+{
+  EZ_ASSERT_DEV(IsInitialized(), "Need to call 'Initialize' first.");
+
+  if (e.m_Type == ezGameApplicationExecutionEvent::Type::BeforeUpdatePlugins)
+  {
+    BeforeUpdatePlugins();
+  }
+}
+
+void ezOpenXR::SetStageSpace(ezXRStageSpace::Enum space)
+{
+  m_StageSpace = space;
+}
+
+void ezOpenXR::SetHMDCamera(ezCamera* pCamera)
+{
+  EZ_ASSERT_DEV(IsInitialized(), "Need to call 'Initialize' first.");
+
+  if (m_pCameraToSynchronize == pCamera)
+    return;
+
+  m_pCameraToSynchronize = pCamera;
+  if (m_pCameraToSynchronize)
+  {
+    m_uiSettingsModificationCounter = m_pCameraToSynchronize->GetSettingsModificationCounter() + 1;
+    m_pCameraToSynchronize->SetCameraMode(ezCameraMode::Stereo, m_pCameraToSynchronize->GetFovOrDim(),
+      m_pCameraToSynchronize->GetNearPlane(), m_pCameraToSynchronize->GetFarPlane());
+  }
+
+  if (ezSoundInterface* pSoundInterface = ezSingletonRegistry::GetSingletonInstance<ezSoundInterface>())
+  {
+    pSoundInterface->SetListenerOverrideMode(m_pCameraToSynchronize != nullptr);
+  }
+}
+
+XrPosef ezOpenXR::ConvertTransform(const ezTransform& tr)
+{
+  XrPosef pose;
+  pose.orientation = ConvertOrientation(tr.m_qRotation);
+  pose.position = ConvertPosition(tr.m_vPosition);
+  return pose;
+}
+
+XrQuaternionf ezOpenXR::ConvertOrientation(const ezQuat& q)
+{
+  return {q.v.y, q.v.z, -q.v.x, -q.w};
+}
+
+XrVector3f ezOpenXR::ConvertPosition(const ezVec3& pos)
+{
+  return {pos.y, pos.z, -pos.x};
+}
+
+ezQuat ezOpenXR::ConvertOrientation(const XrQuaternionf& q)
+{
+  return {-q.z, q.x, q.y, -q.w};
+}
+
+ezVec3 ezOpenXR::ConvertPosition(const XrVector3f& pos)
+{
+  return {-pos.z, pos.x, pos.y};
+}
+
+ezMat4 ezOpenXR::ConvertPoseToMatrix(const XrPosef& pose)
+{
+  ezMat4 m;
+  ezMat3 rot = ConvertOrientation(pose.orientation).GetAsMat3();
+  ezVec3 pos = ConvertPosition(pose.position);
+  m.SetTransformationMatrix(rot, pos);
+  return m;
+}
+
+ezGALResourceFormat::Enum ezOpenXR::ConvertTextureFormat(int64_t format)
+{
+  switch (format)
+  {
+    case DXGI_FORMAT_R8G8B8A8_UNORM_SRGB:
+      return ezGALResourceFormat::RGBAUByteNormalizedsRGB;
+    case DXGI_FORMAT_B8G8R8A8_UNORM_SRGB:
+      return ezGALResourceFormat::BGRAUByteNormalizedsRGB;
+    case DXGI_FORMAT_D32_FLOAT:
+      return ezGALResourceFormat::DFloat;
+    case DXGI_FORMAT_D16_UNORM:
+      return ezGALResourceFormat::D16;
+    case DXGI_FORMAT_D24_UNORM_S8_UINT:
+      return ezGALResourceFormat::D24S8;
+    default:
+      EZ_REPORT_FAILURE("Unknown format {}", format);
+      break;
+  }
+  return ezGALResourceFormat::Invalid;
+}
+
+ezGALMSAASampleCount::Enum ezOpenXR::ConvertMSAACount(ezUInt32 count)
+{
+  switch (count)
```

Comment:
The enums have the right values, you can literally do an int cast.

Reply:
Ok will replace it with static assert that this enum remains that way.